docs/docsite/rst/dev_guide/developing_modules_general_windows.rst

```diff
@@ -0,0 +1,261 @@
+Windows Ansible Module Development Walkthrough
+==============================================
+
+In this section, we will walk through developing, testing, and debugging an
+Ansible Windows module.
+
+Because Windows modules are written in Powershell and needs to be run on a
+Windows host, this guide differs from the usual development walkthrough guide.
+
+What's covered in this section:
+
+.. contents:: Topics
+
+
+Windows environment setup
+=========================
+
+TODO: Add in more information on how to use Vagrant to setup a Windows host.
+
+
+Windows new module development
+==============================
+
+When creating a new module there are a few things to keep in mind
+
+- Module code are in Powershell (.ps1) files while the documentation are in Python (.py) files of the same name
+- Avoid using ``Write-Host/Debug/Verbose/Error`` in the module and add what needs to be returned to the ``$result`` variable
+- When trying an exception use ``Fail-Json -obj $result -message "exception message here"`` instead
+- Most new modules require check mode and integration tests before they are merged into the main Ansible codebase
+- Avoid using try/catch statements over a bit code block, rather use then for individual calls so the error message can be more descriptive
+- Avoid using PSCustomObjects unless necessary
+- Look for common functions at ``./lib/ansible/module_utils/powershell/`` and use the code there instead of duplicating work, these can be imported by adding the line ``#Requires -Module *`` below ``#POWERSHELL_COMMON`` where * is the filename to import
```

Comment:
"...common functions in" instead of "common functions at"; add period after "work" and start a new sentence.